app/models/mixins/supports_feature_mixin.rb

```diff
@@ -0,0 +1,56 @@
+module SupportsFeatureMixin
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def supports(feature, is_supported = true, reason = "", &block)
+      method_name = "supports_#{feature}?"
+
+      self.class.send(:define_method, method_name) do
+        result = if is_supported
+                   true
+                 else
+                   reason
+                 end
+        if result === true
+          true
+        else
+          false
+        end
```

Comment:
Yup.

This stems from my comment [here](https://github.com/ManageIQ/manageiq/pull/8886#issuecomment-221301325). It's intended to strong compare `=== true` as opposed to 'truthy' (weird). I suggest we separate the concern of whether or not something is supported (bool) and the reason why it isn't. This involves returning either a custom result object (make separate class or Struct of some variety) or at minimum a multivalue array (`[true, nil]`, `[false, "Reason why it's not supported"]`)